templates/addons/etcd-backup-restore/etcd-backup-restore.yaml

```diff
@@ -0,0 +1,135 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: ${CLUSTER_NAME}-etcd-backup-restore-addon
+data:
+  etcd-backup-restore-statefulset.yaml: |
+    ---
+    apiVersion: apps/v1
+    kind: StatefulSet
+    metadata:
+      name: etcd-backup
+      namespace: kube-system
+    spec:
+      selector:
+        matchLabels:
+          app.kubernetes.io/name: etcd
+      updateStrategy:
+        type: RollingUpdate
+      template:
+        metadata:
+          labels:
+            app.kubernetes.io/name: etcd
+        spec:
+          initContainers:
+          - name: create-creds-file
+            image: busybox
+            command:
+            - ash
+            - -c
+            - |
+              cat << EOF > /data/credentials-file
+              {
+                "accessKeyID": "$${AWS_ACCESS_KEY_ID}",
+                "secretAccessKey": "$${AWS_SECRET_ACCESS_KEY}",
+                "region": "$${AWS_REGION}",
+                "endpoint": "$${AWS_REGION}.linodeobjects.com",
+                "sseCustomerKey": "$${AWS_SSE_CUSTOMER_KEY}"
+              }
+              EOF
+            volumeMounts:
+              - name: data-volume
+                mountPath: /data
+            env:
+            - name: "AWS_REGION"
+              value: ${OBJ_BUCKET_REGION}
+            - name: "AWS_ACCESS_KEY_ID"
+              valueFrom:
+                secretKeyRef:
+                  name: etcd-backup
+                  key: "access_key_rw"
+            - name: "AWS_SECRET_ACCESS_KEY"
+              valueFrom:
+                secretKeyRef:
+                  name: etcd-backup
+                  key: "secret_key_rw"
+            - name: "AWS_SSE_CUSTOMER_KEY"
+              valueFrom:
+                secretKeyRef:
+                  name: sse-key
+                  key: sse-key
+          containers:
+          - name: backup-restore
+            command:
+            - /etcdbrctl
+            - snapshot
+            - --schedule=0 0 * * *
```

Comment:
do we want to make these schedules optional env variables to start that default to these crons?

Reply:
done